### PR TITLE
What does travis say now

### PR DESCRIPTION
**Pull request by condret**

### CODE REVIEW

File: libr/core/cmd_open.c [hunk 1]
```diff
@@ -513,20 +597,45 @@ static int cmd_open(void *data, const char *input) {
...
 			}
-			ptr = strchr (fn, ' ');
-			if (ptr) {
-				*ptr++ = '\0';
-				char *ptr2 = strchr (ptr, ' ');
-				if (ptr2) {
-					*ptr2++ = 0;
-					ba = r_num_math (core->num, ptr2);
+			ptr = strchr (input+(isn?2:1), ' ');
+			if (ptr && ptr[1]=='0' && ptr[2]=='x') { // hack to fix opening files with space in path
```
Comment (j123123) [inline libr/core/cmd_open.c]: codestyle

File: libr/core/cmd_open.c [hunk 2]
```diff
@@ -570,7 +679,7 @@ static int cmd_open(void *data, const char *input) {
 		break;
 	case '-': // o-
 		switch (input[1]) {
-		case '*': // "o-*"
+			case '*': // "o-*"
```
Comment (j123123) [inline libr/core/cmd_open.c]: codestyle

File: libr/io/buffer.c [hunk 3]
```diff
@@ -3,27 +3,21 @@
...
 
 R_API int r_io_buffer_load(RIO* io, ut64 addr, int len) {
-	ut64 at;
-	int i, r;
 	ut8 buf[512];
-	if (len < 1) {
-		return false;
-	}
+	int i;
+	if (len<1) return false;
```
Comment (j123123) [inline libr/io/buffer.c]: codestyle

File: libr/io/buffer.c [hunk 4]
```diff
@@ -3,27 +3,21 @@
...
 	io->buffer_enabled = 0;
-	for (i = 0; i < len; i += sizeof (buf)) {
-		at = addr + i;
-		(void) r_io_seek (io, at, R_IO_SEEK_SET);
+	for (i=0; i<len; i+=sizeof (buf)) {
```
Comment (j123123) [inline libr/io/buffer.c]: codestyle

File: libr/io/cache.c [hunk 5]
```diff
@@ -177,7 +137,7 @@ R_API int r_io_cache_read(RIO *io, ut64 addr, ut8 *buf, int len) {
 			} else if (ret<0) {
 				da = 0;
 				db = -ret;
-				l = c->size - db;
+				l = c->size-db;
```
Comment (j123123) [inline libr/io/cache.c]: codestyle

File: libr/io/p/io_malloc.c [hunk 6]
```diff
@@ -110,11 +109,9 @@ static bool __check(RIO *io, const char *pathname, bool many) {
 }
 
 static RIODesc *__open(RIO *io, const char *pathname, int rw, int mode) {
-	if (__check (io, pathname, 0)) {
-		RIOMalloc *mal = R_NEW0 (RIOMalloc);
-		if (!mal) return NULL;
-		rw = 7; // RWX
-		mal->fd = -2; /* causes r_io_desc_new() to set the correct fd */
+	if (__check (io, pathname,0)) {
+		RIOMalloc *mal = R_NEW (RIOMalloc);
```
Comment (j123123) [inline libr/io/p/io_malloc.c]: NULL check?

File: libr/io/map.c [hunk 7]
```diff
@@ -1,215 +1,283 @@
...
 
-R_API RIOMap * r_io_map_new(RIO *io, int fd, int flags, ut64 delta, ut64 addr, ut64 size) {
-	RIOMap *map = R_NEW0 (RIOMap);
-	if (!map || ((UT64_MAX - size) < addr)) { //prevent interger-overflow
+R_API RIOMap *r_io_map_new (RIO *io, int fd, int flags, ut64 delta, ut64 addr, ut64 size)
+{
+	RIOMap *map = NULL;
+	if (!size || !io || !io->maps || ((UT64_MAX - size + 1) < addr))			//prevent overflow
+		return NULL;
+	map = R_NEW0 (RIOMap);
```
Comment (j123123) [inline libr/io/map.c]: NULL check?

File: libr/bin/bin.c [hunk 8]
```diff
@@ -882,70 +877,33 @@ R_API int r_bin_load_io_at_offset_as_sz (RBin *bin, RIODesc *desc,
...
-			}
-			sz = totalsz;
-		} else {
-			free (buf_bytes);
-			iob->desc_seek (io, desc, seekaddr);
-			buf_bytes = iob->desc_read (io, desc, &sz);
-		}
+        	ut64 seekaddr = is_debugger? baseaddr: loadaddr;
+        	if (seekaddr == UT64_MAX) seekaddr = 0;
+		buf_bytes = malloc (sz);
```
Comment (j123123) [inline libr/bin/bin.c]: NULL check?

File: libr/io/p/io_w32dbg.c [hunk 9]
```diff
@@ -73,8 +73,8 @@ static RIODesc *__open(RIO *io, const char *file, int rw, int mode) {
 			return NULL;
 		}
 		pidpath = r_sys_pid_to_path (dbg->pid);
-		RETURN_IO_DESC_NEW (&r_io_plugin_w32dbg, -1,
-			pidpath, rw | R_IO_EXEC, mode, dbg);
+		return r_io_desc_new (io, &r_io_plugin_w32dbg,
+				pidpath, rw | R_IO_EXEC, mode, dbg);
```
Comment (j123123) [inline libr/io/p/io_w32dbg.c]: codestyle

File: libr/io/plugin.c [hunk 10]
```diff
@@ -2,93 +2,98 @@
...
+	if (!io)
+		return false;
+	io->plugins = ls_new ();
+	io->plugins->free = free;
 
-	r_list_free (io->plugins);
-	io->plugins = r_list_newf (free);
-	for (i = 0; io_static_plugins[i]; i++) {
-		if (!io_static_plugins[i]->name) {
+	for (i=0; io_static_plugins[i]; i++) {
```
Comment (j123123) [inline libr/io/plugin.c]: codestyle

File: libr/core/cio.c [hunk 11]
```diff
@@ -325,24 +325,12 @@ R_API bool r_core_seek(RCore *core, ut64 addr, bool rb) {
...
+			core->offset = old;
 		}
 	}
-	if (core->section != newsection) {
-		r_core_seek_archbits (core, core->offset);
-		core->section = newsection;
-	}
-	return (ret == -1)? false: true;
+	r_core_seek_archbits (core, core->offset);
+	return (ret==-1)? false: true;
```
Comment (j123123) [inline libr/core/cio.c]: codestyle

File: libr/core/cmd_search.c [hunk 12]
```diff
@@ -82,9 +82,9 @@ static int search_hash(RCore *core, const char *hashname, const char *hashstr, u
 			eprintf ("Search in range 0x%08"PFMT64x" and 0x%08"PFMT64x"\n", from, to);
 			int blocks = (int)(to - from - len);
 			eprintf ("Carving %d blocks...\n", blocks);
-			(void)r_io_read_at (core->io, from, buf, bufsz);
-			for (i = 0; (from + i + len) < to; i++) {
-				char *s = r_hash_to_string (NULL, hashname, buf + i, len);
+			r_io_read_at (core->io, from, buf, bufsz);
+			for (i = 0; (from+i+len)<to; i++) {
```
Comment (j123123) [inline libr/core/cmd_search.c]: codestyle

File: libr/core/core.c [hunk 13]
```diff
@@ -491,16 +497,18 @@ static ut64 num_callback(RNum *userptr, const char *str, int *ok) {
...
 		case 'S':
-			if ((s = r_io_section_vget (core->io, core->offset))) {
-				return (str[2] == 'S'? s->size: s->vaddr);
+			{
+				SdbList *secs = r_io_section_vget_secs_at (core->io, core->offset);
+				s = secs ? ls_pop (secs) : NULL;
+				ls_free (secs);
 			}
-			return 0LL;
+			return s? (str[2]=='S'? s->size: s->vaddr): 0LL;
```
Comment (j123123) [inline libr/core/core.c]: codestyle

File: libr/io/cache.c [hunk 14]
```diff
@@ -80,36 +81,22 @@ R_API int r_io_cache_list(RIO *io, int rad) {
...
 			io->cb_printf (" # replaces: ");
-		  	for (i = 0; i < c->size; i++) {
-				io->cb_printf ("%02x", (ut8)(c->odata[i] & 0xff));
-			}
-			io->cb_printf ("\n");
-		} else if (rad == 2) {
-			io->cb_printf ("{\"idx\":%"PFMT64d",\"addr\":%"PFMT64d",\"size\":%d,", j, c->from, c->size);
-			io->cb_printf ("\"before\":\"");
-		  	for (i = 0; i < c->size; i++) {
+			for (i=0; i<c->size; i++)
```
Comment (j123123) [inline libr/io/cache.c]: codestyle

File: libr/io/cache.c [hunk 15]
```diff
@@ -80,36 +81,22 @@ R_API int r_io_cache_list(RIO *io, int rad) {
...
-			}
-			io->cb_printf ("\",\"written\":%s}%s", c->written? "true": "false",iter->n? ",": "");
-		} else if (rad == 0) {
-			io->cb_printf ("idx=%d addr=0x%08"PFMT64x" size=%d ", j, c->from, c->size);
-			for (i = 0; i < c->size; i++) {
+			io->cb_printf ("\n");
+		} else {
+			io->cb_printf ("idx=%d addr=0x%08"PFMT64x" size=%d ",
+				j, c->from, c->size);
+			for (i=0; i<c->size; i++)
```
Comment (j123123) [inline libr/io/cache.c]: codestyle

File: libr/io/cache.c [hunk 16]
```diff
@@ -80,36 +81,22 @@ R_API int r_io_cache_list(RIO *io, int rad) {
...
 				io->cb_printf ("%02x", c->odata[i]);
-			}
 			io->cb_printf (" -> ");
-			for (i = 0; i < c->size; i++) {
+			for (i=0; i<c->size; i++)
```
Comment (j123123) [inline libr/io/cache.c]: codestyle

File: binr/rahash2/rahash2.c [hunk 17]
```diff
@@ -219,9 +219,9 @@ static int do_hash(const char *file, const char *algo, RIO *io, int bsize, int r
 				oto = to;
 				f = from;
 				t = to;
-				for (j = f; j < t; j += bsize) {
-					int nsize = (j + bsize < fsize) ? bsize : (fsize - j);
-					r_io_pread (io, j, buf, bsize);
+				for (j=f; j<t; j+=bsize) {
```
Comment (condret) [inline binr/rahash2/rahash2.c]: hä? i don't see anything wrong here

Comment (j123123) [inline binr/rahash2/rahash2.c]: ``j = f``
and
``j=f``

---
``j<t``
and
``j < t``

---
``j+=bsize``
and
``j += bsize``

https://github.com/radare/radare2/blob/master/CONTRIBUTING.md
>Use a space after most of the keyword and around operators.

Comment (condret) [inline binr/rahash2/rahash2.c]: k :/

**Comment by condret** [comment]
well, from my perspective, this is ready for merge. Yes it has a few bugs, but basically everything works. I guess most of the XXX in the testsuite come from new output formats. The only point for not merging is the WINAPI-stuff

**Comment by radare** [comment]
I have to test first and i want it rebased on top of master after greenifying travis. This is goong to be fun

File: libr/io/cache.c (hunk 14 above)
Comment (radare) [inline libr/io/cache.c]: use r2-indent

**Comment by radare** [comment]
conflicts!

**Comment by radare** [comment]
closing this pr as long as real work is done in the opsiol branch

**Pull request closed without merging**